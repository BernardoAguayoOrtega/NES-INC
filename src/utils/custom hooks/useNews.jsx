// use react hooks
import { useEffect, useState } from 'react';
// import request tools
import { newsAPI, callNewsApi } from '../requests.js';

const useNews = () => {
  // use state to handle the state
  const [news, setNews] = useState({
    data: [{ title: '', images: '' }],
    loading: true,
  });

  useEffect((option = 0) => {
    // set implementation
    switch (option) {
      case 0: {
        const getInitialNews = callNewsApi('news');

        const fetchData = () =>
          newsAPI
            .get(getInitialNews('?page=10'))
            .then((response) =>
              setNews({
                data: response?.data.data,
                info: response?.data.info,
                loading: false,
              }),
            );

        fetchData();
      }
    }
  }, []);

  return { news };
};

export default useNews;
