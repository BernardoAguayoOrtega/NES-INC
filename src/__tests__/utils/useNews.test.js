// import app component
import useNews from '../../utils/custom hooks/useNews.jsx';

describe('request.js', () => {

  it('is match snapshot newsAPI?', () => {
    expect(useNews)
      .toMatchSnapshot()
  });

});
