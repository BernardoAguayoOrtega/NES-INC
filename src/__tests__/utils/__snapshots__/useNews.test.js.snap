// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request.js is match snapshot newsAPI? 1`] = `[Function]`;
